source/dub/packagemanager.d

```diff
@@ -445,138 +494,126 @@ class PackageManager {
 		logInfo("Removed package: '"~pack.name~"'");
 	}
 
-	Package addLocalPackage(Path path, string verName, LocalPackageType type)
+	deprecated Package addLocalPackage(Path path, string verName, LocalPackageType type)
 	{
-		path.endsWithSlash = true;
-		auto pack = Package.load(path);
-		enforce(pack.name.length, "The package has no name, defined in: " ~ path.toString());
-		if (verName.length)
-			pack.version_ = Version(verName);
-
-		// don't double-add packages
-		Package[]* packs = &m_repositories[type].localPackages;
-		foreach (p; *packs) {
-			if (p.path == path) {
-				enforce(p.version_ == pack.version_, "Adding the same local package twice with differing versions is not allowed.");
-				logInfo("Package is already registered: %s (version: %s)", p.name, p.version_);
-				return p;
-			}
-		}
-
-		addPackages(*packs, pack);
-
-		writeLocalPackageList(type);
-
-		logInfo("Registered package: %s (version: %s)", pack.name, pack.version_);
-		return pack;
+		return addLocalPackage(path, verName, m_defaultPaths[type]);
+	}
+	Package addLocalPackage(Path path, string verName, Path repoPath)
+	in { assert(repoPath.absolute); }
+	body
+{
+		repoPath.normalize();
+		return m_repos[repoPath].addLocalPackage(path, verName);
 	}
 
-	void removeLocalPackage(Path path, LocalPackageType type)
```

Comment:
The work is now done as member of Repository, but there's a Path-based function here to refer to the right member of m_repos.